docs/key_concepts.rst

```diff
@@ -0,0 +1,83 @@
+Key Concepts
+============
+
+This section introduces nameko's central concepts.
+
+Anatomy of a Service
+--------------------
+
+A nameko service is just a Python class. The class encapsulates the application logic in its methods and declares any :ref:`dependencies <dependencies>` as attributes.
+
+Methods are exposed to the outside world with :ref:`entrypoint <entrypoints>` decorators.
+
+.. literalinclude:: examples/example.py
```

Comment:
I think it would help if you could extend this example with another block of code showing a standalone rpc proxy calling the rpc method, and then you can add comments pinpointing "this is a dependency", "this is an entrypoint", "here we are instatiating a worker". I think it helps to see each of the concepts "in action".